### DISCUSSION TITLE
GitHub Actions for CB-SA

### DISCUSSION THREAD

**Discussion by pvperez1** [CollectionBuilder/collectionbuilder.github.io#34] [Q&A] [answered]
Hello, first of all, I would like to thank the whole team for this great project! CB has been really helpful to our organization.

Currently, we are using the CB-Stand Alone version and every time we updated something, we copy-paste the contents of _site to our server. I think I saw a guide somewhere created by the team on how to automate this process through GitHub Actions. We have not used GitHub Actions before, and what we want to achieve is that every time we push the commits to GitHub, it will be reflected on the site, without doing the manual copy-paste.

Is this possible? I would appreciate any links to resources that will help us do that! Thank you again!

**Comment by evanwill** [comment, marked as answer]
@pvperez1 thanks for reaching out! 
Yes it is possible to automate the build using GitHub Actions (or other 3rd party platforms). 

Here is the doc link for GitHub Actions, https://collectionbuilder.github.io/cb-docs/docs/deploy/actions/
and for Render, https://collectionbuilder.github.io/cb-docs/docs/deploy/thirdparty/
These will get your site automatically built each time you push something new or create a PR on the main branch.

By default the GitHub Actions version will be hosted at username.github.io/repositoryname (just like using GH Pages), and Render will let you choose a free subdomain as well. So the next step, which isn't currently documented on cb-docs, is to set up a custom domain. If you own your domain, it isn't too hard. Check the GitHub docs for details, https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/about-custom-domains-and-github-pages

Let us know how it goes!

**Reply by pvperez1** [reply]
Hi @evanwill, thank you so much! Both guides worked but we ended up using Render because it allowed us to choose a specific repository from github and our own previously bought domain, all for free. The limitation with Github Actions is that the custom domain will be for the whole github account or organization, which is not our set-up as of the moment.
Thanks again!
